.evergreen/generated_configs/functions.yml

```diff
@@ -70,6 +70,90 @@ functions:
         args:
           - -c
           - .evergreen/scripts/check-preludes.py .
+  cse-sasl-auto-darwinssl-compile:
+    - command: expansions.update
+      params:
+        updates:
+          - { key: SSL, value: DARWIN }
+          - { key: SASL, value: AUTO }
+    - command: subprocess.exec
+      type: test
+      params:
+        binary: bash
+        working_dir: mongoc
+        add_expansions_to_env: true
+        env:
+          COMPILE_LIBMONGOCRYPT: "ON"
+        args:
+          - -c
+          - EXTRA_CONFIGURE_FLAGS="-DENABLE_PIC=ON -DENABLE_CLIENT_SIDE_ENCRYPTION=ON ${EXTRA_CONFIGURE_FLAGS}" .evergreen/scripts/compile.sh
+  cse-sasl-auto-openssl-compile:
+    - command: expansions.update
+      params:
+        updates:
+          - { key: SSL, value: OPENSSL }
+          - { key: SASL, value: AUTO }
+    - command: subprocess.exec
+      type: test
+      params:
+        binary: bash
+        working_dir: mongoc
+        add_expansions_to_env: true
+        env:
+          COMPILE_LIBMONGOCRYPT: "ON"
+        args:
+          - -c
+          - EXTRA_CONFIGURE_FLAGS="-DENABLE_PIC=ON -DENABLE_CLIENT_SIDE_ENCRYPTION=ON ${EXTRA_CONFIGURE_FLAGS}" .evergreen/scripts/compile.sh
+  cse-sasl-auto-openssl-static-compile:
+    - command: expansions.update
+      params:
+        updates:
+          - { key: SSL, value: OPENSSL_STATIC }
+          - { key: SASL, value: AUTO }
+    - command: subprocess.exec
+      type: test
+      params:
+        binary: bash
+        working_dir: mongoc
+        add_expansions_to_env: true
+        env:
+          COMPILE_LIBMONGOCRYPT: "ON"
+        args:
+          - -c
+          - EXTRA_CONFIGURE_FLAGS="-DENABLE_PIC=ON -DENABLE_CLIENT_SIDE_ENCRYPTION=ON ${EXTRA_CONFIGURE_FLAGS}" .evergreen/scripts/compile.sh
+  cse-sasl-auto-winssl-compile:
+    - command: expansions.update
+      params:
+        updates:
+          - { key: SSL, value: WINDOWS }
+          - { key: SASL, value: AUTO }
+    - command: subprocess.exec
+      type: test
+      params:
+        binary: bash
+        working_dir: mongoc
+        add_expansions_to_env: true
+        env:
+          COMPILE_LIBMONGOCRYPT: "ON"
+        args:
+          - -c
+          - EXTRA_CONFIGURE_FLAGS="-DENABLE_PIC=ON -DENABLE_CLIENT_SIDE_ENCRYPTION=ON ${EXTRA_CONFIGURE_FLAGS}" .evergreen/scripts/compile.sh
+  cse-sasl-off-openssl-compile:
+    - command: expansions.update
+      params:
+        updates:
+          - { key: SSL, value: OPENSSL }
```

Comment:
I'm not sure what's going on here, but [this task](https://spruce.mongodb.com/task/mongo_c_driver_sanitizers_matrix_asan_asan_cse_sasl_off_openssl_ubuntu1804_clang_compile_patch_cb616d9b0b95ac5d2a0a464c9ec5befcff3ccd93_63dd69832fbabe50d8864a33_23_02_03_20_07_33/logs?execution=0) still seems to be compiling with SASL, rather than without it:

```
[2023/02/03 20:09:30.334] -- Searching for sasl/sasl.h
[2023/02/03 20:09:30.334] --   Found in /usr/include
[2023/02/03 20:09:30.334] -- Searching for libsasl2
[2023/02/03 20:09:30.336] --   Found /usr/lib/x86_64-linux-gnu/libsasl2.so
[2023/02/03 20:09:30.336] -- Looking for sasl_client_done
[2023/02/03 20:09:30.527] -- Looking for sasl_client_done - found
[2023/02/03 20:09:30.527] -- Check size of socklen_t
[2023/02/03 20:09:30.527] -- Check size of socklen_t - done
[2023/02/03 20:09:30.611] -- Looking for res_nsearch
[2023/02/03 20:09:30.611] -- Looking for res_nsearch - found
[2023/02/03 20:09:30.611] -- Looking for res_ndestroy
[2023/02/03 20:09:30.743] -- Looking for res_ndestroy - not found
[2023/02/03 20:09:30.743] -- Looking for res_nclose
[2023/02/03 20:09:30.743] -- Looking for res_nclose - found
[2023/02/03 20:09:30.856] -- Looking for sched_getcpu
[2023/02/03 20:09:30.856] -- Looking for sched_getcpu - not found
[2023/02/03 20:09:30.856] -- Detected parameters: accept (int, struct sockaddr *, socklen_t *)
[2023/02/03 20:09:30.857] -- Searching for compression library header snappy-c.h
[2023/02/03 20:09:30.857] --   Found in /usr/include
[2023/02/03 20:09:30.857] -- Searching for libsnappy
[2023/02/03 20:09:30.865] --   Found /usr/lib/x86_64-linux-gnu/libsnappy.so
[2023/02/03 20:09:30.865] -- Performing Test MONGOC_HAVE_SS_FAMILY
[2023/02/03 20:09:30.941] -- Performing Test MONGOC_HAVE_SS_FAMILY - Success
[2023/02/03 20:09:30.941] -- Compiling against OpenSSL
[2023/02/03 20:09:30.941] -- Compiling against Cyrus SASL
```

I am not positive that this output corresponds to this particular task definition, but the task name is `*-sasl-off-*`, which implies that it should be compiling without SASL.

Reply:
This output is coming from libmongocrypt compiling libmongoc (for tests) alongside libbson (what it wants). The output is [also present](https://parsley.mongodb.com/evergreen/mongo_c_driver_asan_ubuntu_debug_compile_asan_openssl_cse_8b5a14e5a35fb7b365a1a53f24c933f8c497f337_23_02_03_22_40_36/0/task?bookmarks=0,2604&shareLine=457) in the [current corresponding task](https://spruce.mongodb.com/task/mongo_c_driver_asan_ubuntu_debug_compile_asan_openssl_cse_8b5a14e5a35fb7b365a1a53f24c933f8c497f337_23_02_03_22_40_36/logs?execution=0). This suggests we can add some environment variables to pass to libmongocrypt to avoid redundant compilation.

Reply:
Well, that's odd and unexpected.  It might be good to consider that sort of improvement in a follow-up PR.